File: 1514-MinimumValueToGetPositiveStepByStepSum/1514-MinimumValueToGetPositiveStepByStepSum.c
// Last updated: 6/6/2026, 10:24:50 PM
#include <stdio.h>

int minStartValue(int *nums, int n) {
    int minsum = 0;
    int psum = 0;
    
    for (int i = 0; i < n; i++) {
        psum += nums[i];
        if (psum < minsum) {
            minsum = psum;
        }
    }
    
    return 1 - minsum;
}

File: 1514-MinimumValueToGetPositiveStepByStepSum/1514-MinimumValueToGetPositiveStepByStepSum.py
# Last updated: 6/6/2026, 10:24:56 PM
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        st = 1
        cal = 0
        while True:
            cal = st
            cnt = 0
            for i in range(len(nums)):
                if cal + nums[i] >= 1:
                    cal = cal + nums[i]
                    cnt += 1
                else:
                    st += 1
                    break
            if cnt == len(nums):
                break
        return st